drivers/clock/arm/timer.c

```diff
@@ -97,12 +98,22 @@ static inline uint64_t freq_cycles_and_hz_to_ns(uint64_t ncycles, uint64_t hz)
         return ncycles * US_IN_S / (hz / KHZ);
     }
 
-    return (ncycles * NS_IN_S) / hz;
+    __uint128_t ncycles_in_s = (__uint128_t)ncycles * NS_IN_S;
+    /* We can discard the remainder. */
+    uint64_t rem = 0;
+    uint64_t res = udiv128by64to64(ncycles_in_s >> 64, ncycles_in_s & 0xffffffffffffffff, NS_IN_S, &rem);
```

Comment:
Make that 0xffffffffffffffffULL, or just cast to a 64-bit number.
`udiv128by64to64(ncycles_in_s >> 64, (uint64_t)ncycyles_in_s, NS_IN_S, &rem)`
or create some helper macros to extract high/low 64-bit words from a 128bit value